frame/uniques/src/functions.rs

```diff
@@ -85,10 +83,22 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		);
 
 		CollectionAccount::<T, I>::insert(&owner, &collection, ());
+
 		Self::deposit_event(event);
 		Ok(())
 	}
 
+	pub fn try_increment_id(
+		last_collection: T::CollectionId,
+	) -> Result<T::CollectionId, Error<T, I>> {
+		let new_collection = last_collection + T::CollectionId::one();
+
+		CollectionsCount::<T, I>::set(new_collection);
+		ensure!(!Collection::<T, I>::contains_key(new_collection), Error::<T, I>::InUse);
```

Comment:
you should do the check before you set the storage.

Reply:
Because if you don't do it like this the ```CollectionsCount``` never gets updated if a used id is encountered. If this happens no matter how many times you call the ```create``` function you will always get the same error.